app/src/main/java/com/github/dfa/diaspora_android/util/OrbotStatusReceiver.java

```diff
@@ -0,0 +1,141 @@
+/*
+    This file is part of the Diaspora for Android.
+
+    Diaspora for Android is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Diaspora for Android is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with the Diaspora for Android.
+
+    If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.github.dfa.diaspora_android.util;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import com.github.dfa.diaspora_android.App;
+import com.github.dfa.diaspora_android.activity.MainActivity;
+import com.github.dfa.diaspora_android.data.AppSettings;
+
+import info.guardianproject.netcipher.NetCipher;
+import info.guardianproject.netcipher.proxy.OrbotHelper;
+import info.guardianproject.netcipher.web.WebkitProxy;
+
+/**
+ * BroadcastReceiver that handles Orbot status intents and sets the proxy.
+ * Created by vanitas on 06.06.16.
+ */
+public class OrbotStatusReceiver extends BroadcastReceiver {
+
+    public static final String EXTRA_HTTP_HOST = "org.torproject.android.intent.extra.HTTP_PROXY_HOST";
+    public static final String EXTRA_HTTP_PORT = "org.torproject.android.intent.extra.HTTP_PROXY_PORT";
+    public static final String defaultHost = "127.0.0.1";
+    public static final int defaultPort = 8118;
+
+    private static String host = "";
+    private static int port = 0;
+    private Intent lastStatus;
+    private Context lastContext;
+    private static MainActivity mainActivity;
```

Comment:
The problem is, that I need the Activity in onReceive because I need to call MainActivity.requestOrbotStart . Do you know a way to get access to this method from within the receiver in a better way?